### PR TITLE
Add nRF52/nRF53/nRF91 and some test boards

### PR DESCRIPTION
**Pull request by gmarull**
TODO:

- [ ] `_ns` boards do not seem to work, needs to be investigated
- [x] If boards are moved to e.g. `tests/boards` (and `zephyr/module.yml` updated with `board_root: tests`), they no longer work
- [x] Device whether we want to keep the `board_root`, likely not. Now present to ease testing.

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]
lgtm.

Including https://github.com/zephyrproject-rtos/zephyr/pull/74057 allows to place boards under `tests/boards`.

File: tests/boards/arm/nrf9160dk_nrf9160_v1/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# nRF9160 DK NRF9160 board configuration
+
+# Copyright (c) 2018-2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NRF9160DK_NRF9160_V1 || BOARD_NRF9160DK_NRF9160_V1_NS
+
+config BOARD
+	default "nrf9160dk_nrf9160"
```
Comment (nordicjm) [inline tests/boards/arm/nrf9160dk_nrf9160_v1/Kconfig.defconfig]: not updated

File: tests/boards/arm/nrf9160dk_nrf9160_v1/nrf9160dk_nrf9160.dts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (nordicjm) [inline tests/boards/arm/nrf9160dk_nrf9160_v1/nrf9160dk_nrf9160.dts]: filename not updated

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: tests/boards/arm/nrf9160dk_nrf9160_v1/Kconfig.defconfig (hunk 1 above)
Comment (gmarull) [reply tests/boards/arm/nrf9160dk_nrf9160_v1/Kconfig.defconfig]: fixed

File: tests/boards/arm/nrf9160dk_nrf9160_v1/nrf9160dk_nrf9160.dts (hunk 2 above)
Comment (gmarull) [reply tests/boards/arm/nrf9160dk_nrf9160_v1/nrf9160dk_nrf9160.dts]: fixed

File: zephyr/module.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+build:
+  settings:
+    soc_root: .
+    board_root: .
```
Comment (nordicjm) [inline zephyr/module.yml]: board root should only be used in tests, not generally available. These boards should not appear in `west boards` or be usable in normal `west build -b...`

Comment (gmarull) [reply zephyr/module.yml]: thx, missed removal

**Review by nordicjm — APPROVED** [review]

**Pull request merged**